src/Srclib/Types.hs

```diff
@@ -112,13 +113,36 @@ textToOriginPath = OriginPath . toString
 --   Manifest?: string;
 -- }
 
+data SourceUnitNoticeFile = SourceUnitNoticeFile
+  { sourceUnitNoticeFilePath :: Text
+  , sourceUnitNoticeFileContents :: Text
+  , sourceUnitNoticeFileCopyrights :: Maybe (NonEmpty Text)
+  }
+  deriving (Eq, Ord, Show)
+
+instance ToJSON SourceUnitNoticeFile where
+  toJSON SourceUnitNoticeFile{..} =
+    object
+      [ "path" .= sourceUnitNoticeFilePath
+      , "contents" .= sourceUnitNoticeFileContents
+      , "copyrights" .= sourceUnitNoticeFileCopyrights
+      ]
+
+instance FromJSON SourceUnitNoticeFile where
+  parseJSON = withObject "SourceUnitNoticeFile" $ \obj ->
+    SourceUnitNoticeFile
+      <$> obj .: "path"
+      <*> obj .: "contents"
+      <*> obj .:? "copyrights"
+
 data FullSourceUnit = FullSourceUnit
   { fullSourceUnitName :: Text
   , fullSourceUnitType :: Text
   , fullSourceUnitTitle :: Maybe Text
   , fullSourceUnitManifest :: Maybe Text
   , fullSourceUnitBuild :: Maybe SourceUnitBuild
   , fullSourceUnitGraphBreadth :: GraphBreadth
+  , fullSourceUnitNoticeFiles :: Maybe (NonEmpty SourceUnitNoticeFile)
```

Comment:
What is the distinction between `Maybe (NonEmpty SourceUnitNoticeFile)` and `[SourceUnitNoticeFile]`? The types are isomorphic I think:

```haskell
toList :: Maybe (NonEmpty a) -> [a]
toList = fromMaybe []

fromList :: [a] -> Maybe (NonEmpty a)
fromList [] = Nothing
fromList (x:xs) = Just (x :| xs)
``` 
I prefer using the just the list because it usually means less error checking. Functions operating on lists usually know what to do with an empty one.

Reply:
I needed to use `Maybe NotEmpty` because themis will not have the `NoticeFiles` entry in its JSON output if there are no NoticeFiles, so I needed the json parser to look like this:

```hs
      <*> obj .:? "NoticeFiles"
```

and I don't think you can have the `?` in there unless there's a `Maybe` in the type declaration for `licenseUnitNoticeFiles` and `sourceUnitNoticeFiles.

If that's not correct, I'm happy to make the change to just `[]`

Reply:
You should be able to do `obj .:? "NoticeFiles" .!= []` I think: https://hackage.haskell.org/package/aeson-2.2.3.0/docs/Data-Aeson.html#v:.-33--61-

Reply:
Ahh. It's the `.! = []` I was missing. Thank you!

I'll make that change